### PR TITLE
Feature/update apr13

### PR DESCRIPTION
**Pull request by cmgas**
## Description
Update with develop on april 13.

Waiting on 
https://github.com/JCSDA-internal/fv3-jedi-data/pull/71
https://github.com/JCSDA-internal/soca/pull/864
https://github.com/JCSDA-internal/fv3-jedi/pull/847

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -157,15 +157,15 @@ if(UFS_APP MATCHES "^(S2S)$")
   # fv3-jedi and associated repositories
   # ------------------------------------
   ecbuild_bundle( PROJECT femps    GIT "https://github.com/jcsda-internal/femps.git"    TAG 1.2.0 )
-  ecbuild_bundle( PROJECT fv3-jedi GIT "https://github.com/jcsda-internal/fv3-jedi.git" BRANCH feature/ufs_dom )  # updated from develop on jan28
-  ecbuild_bundle( PROJECT soca     GIT "https://github.com/jcsda-internal/soca.git"     BRANCH feature/ufs_dom )  # updated from develop on jan28
+  ecbuild_bundle( PROJECT fv3-jedi GIT "https://github.com/jcsda-internal/fv3-jedi.git" BRANCH feature/ufs_clem_update )  # updated from develop on jan28
```
Comment (shlyaeva) [inline CMakeLists.txt]: This could be updated to `feature/ufs_dom` once fv3-jedi PR is merged

File: CMakeLists.txt [hunk 2]
```diff
@@ -157,15 +157,15 @@ if(UFS_APP MATCHES "^(S2S)$")
...
+  ecbuild_bundle( PROJECT soca     GIT "https://github.com/jcsda-internal/soca.git"     BRANCH feature/ufs_clem )  # updated from develop on jan28
```
Comment (shlyaeva) [inline CMakeLists.txt]: This branch is not pushed to remote, I haven't tested this.

**Review by shlyaeva — COMMENTED** [review]
Tested for `ATM`, works as expected. soca branch is missing on the remote, so I haven't tested the rest yet.

**Comment by climbfuji** [comment]
Testing this now!

File: CMakeLists.txt (hunk 2 above)
Comment (cmgas) [reply CMakeLists.txt]: I am pushing it now

**Review by shlyaeva — DISMISSED** [review]
couldn't test with soca on platform/compiler of choice, approving based on tests with ATM.

**Comment by climbfuji** [comment]
This doesn't work for me :-( On my Mac, I get the following error when I run the UFS ctests (warmstart):
```
1197: At line 47 of file /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13/fv3-jedi/src/fv3jedi/Utilities/fv3jedi_constants_mod.f90
1197: Fortran runtime error: Different CHARACTER lengths (7/1) in array constructor
1197: At line 47 of file /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13/fv3-jedi/src/fv3jedi/Utilities/fv3jedi_constants_mod.f90
1197: Fortran runtime error: Different CHARACTER lengths (7/1) in array constructor
1197: At line 47 of file /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13/fv3-jedi/src/fv3jedi/Utilities/fv3jedi_constants_mod.f90
1197: Fortran runtime error: Different CHARACTER lengths (7/1) in array constructor
1197: At line 47 of file /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13/fv3-jedi/src/fv3jedi/Utilities/fv3jedi_constants_mod.f90
1197: Fortran runtime error: Different CHARACTER lengths (7/1) in array constructor
1197: At line 47 of file /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13/fv3-jedi/src/fv3jedi/Utilities/fv3jedi_constants_mod.f90
1197: Fortran runtime error: Different CHARACTER lengths (7/1) in array constructor
1197: At line 47 of file /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13/fv3-jedi/src/fv3jedi/Utilities/fv3jedi_constants_mod.f90
1197: Fortran runtime error: Different CHARACTER lengths (7/1) in array constructor
1197: [JCSDA-L-18146:79842] *** Process received signal ***
```

**Review by climbfuji — CHANGES_REQUESTED** [review]
We'll need to fix the ufs warmstart error in fv3-jedi code on macOS

**Comment by cmgas** [comment]
Rebuilding now with latest develops

**Comment by climbfuji** [comment]
@cmgas This works now! The only thing that's needed on my macOS is updating the reference value for the ufs warmstart test:
```
difference between 355.7572273542616 and 356.73198629080974 is 0.97475893654814172, i.e. exceeds 0.035673198629080978 (FAIL)
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -42,23 +42,27 @@ option("BUNDLE_SKIP_FCKIT" "Don't build fckit" "ON") # Skip fckit build unless u
...
 endif()
 
-# Core JEDI repositories
-# ----------------------
-ecbuild_bundle( PROJECT oops  GIT "https://github.com/jcsda-internal/oops.git"  TAG 0932772f4b20b0defedfb12e31a0c17c18caac71 ) # develop on feb 1
-ecbuild_bundle( PROJECT vader GIT "https://github.com/jcsda-internal/vader.git" TAG 3c8fc58ed67c013a955ca7d8f8d2346f2844ac96 ) # develop on jan 28
-ecbuild_bundle( PROJECT saber GIT "https://github.com/jcsda-internal/saber.git" TAG 1203356523b123654a8871846a55a06ed26b7247 ) # develop on jan 28
-ecbuild_bundle( PROJECT ioda  GIT "https://github.com/jcsda-internal/ioda.git"  TAG 39ab4dd1dfe737f71385af802c2b5c67c2ef8efc ) # develop on jan 28
-ecbuild_bundle( PROJECT ufo   GIT "https://github.com/jcsda-internal/ufo.git"   TAG ab106a84c770414a3edf9dacabeadea5d70d9edf ) # develop on jan 28
+ecbuild_bundle( PROJECT crtm  GIT "https://github.com/jcsda-internal/crtm.git"  TAG v2.4-jedi.2 )
```
Comment (cmgas) [inline CMakeLists.txt]: Should crtm be built here? When I loaded it as a module I couldn't get it to work, but it might have been an error on my end

Comment (climbfuji) [reply CMakeLists.txt]: I didn't realize that ... will remove the line and try again

Comment (climbfuji) [reply CMakeLists.txt]: No, you don't need it, builds just fine with v2.4-jedi.2 from spack-stack

Comment (cmgas) [reply CMakeLists.txt]: I had issues with failing tests

Comment (climbfuji) [reply CMakeLists.txt]: which tests?

Comment (cmgas) [reply CMakeLists.txt]: A dozen or so (`fv3jedi_test_tier1_hofx_fv3lm` for example), all failing with
```
60:  SpcCoeff_ReadFile(Binary)(FAILURE) : File Data/crtm/amsua_n19.SpcCoeff.bin not found.
60:  CRTM_SpcCoeff_Load(FAILURE) : Error reading SpcCoeff file #1, Data/crtm/amsua_n19.SpcCoeff.bin
```

I guess I had an issue when I built the bundle

Comment (climbfuji) [reply CMakeLists.txt]: I am running all 1544 ctests now, will let you know what I find

**Comment by climbfuji** [comment]
Other good news: I was able to build the atmaero configuration on my macOS as well, will try ng-godas next

File: CMakeLists.txt (hunk 3 above)
Comment (climbfuji) [reply CMakeLists.txt]: I've got only these `fv3jedi_*` test failures, but keep in mind that I am using gsibec@1.1.2 which is the reason for some of them to fail:
```
	1189 - fv3jedi_test_tier1_model_ufs_coldstart (Failed)
	1194 - fv3jedi_test_tier1_forecast_ufs_coldstart (Failed)
	1214 - fv3jedi_test_tier1_dirac_geos_gsi_global (Failed)
	1215 - fv3jedi_test_tier1_3dvar_geos_sondes (Failed)
	1216 - fv3jedi_test_tier1_3dvar_geos_ozone (Failed)
	1217 - fv3jedi_test_tier1_dirac_gfs_gsi_global (Failed)
	1218 - fv3jedi_test_tier1_3dvar_gfs_sondes (Failed)
	1236 - fv3jedi_test_tier1_4denvar (Failed)
```

Comment (climbfuji) [reply CMakeLists.txt]: I get many many segfaults after successfully completing the tests for `qg` like this one:
```
...
171: OOPS_STATS Run end                                  - Runtime:      2.57 sec,  Memory: total:    26.99 Mb, per task: min =    26.99 Mb, max =    26.99 Mb
171: Run: Finishing oops::Forecast<QG> with status = 0
171: [TestReference] Comparison is done
171: OOPS Ending   2023-04-19 14:33:36 (UTC-0600)
3/3 Test #171: test_qg_truth_pert_heat ..........***Exception: SegFault  2.92 sec
```
But I think this is a mac specific issue that has to do with the openmp library

Comment (climbfuji) [reply CMakeLists.txt]: So yes, please remove the crtm line here

File: CMakeLists.txt [hunk 4]
```diff
@@ -157,15 +161,15 @@ if(UFS_APP MATCHES "^(S2S)$")
   # fv3-jedi and associated repositories
   # ------------------------------------
   ecbuild_bundle( PROJECT femps    GIT "https://github.com/jcsda-internal/femps.git"    TAG 1.2.0 )
-  ecbuild_bundle( PROJECT fv3-jedi GIT "https://github.com/jcsda-internal/fv3-jedi.git" BRANCH feature/ufs_dom )  # updated from develop on jan28
-  ecbuild_bundle( PROJECT soca     GIT "https://github.com/jcsda-internal/soca.git"     BRANCH feature/ufs_dom )  # updated from develop on jan28
+  ecbuild_bundle( PROJECT fv3-jedi GIT "https://github.com/jcsda-internal/fv3-jedi.git" BRANCH feature/ufs_clem_update )  # updated from develop on apr 18
```
Comment (climbfuji) [inline CMakeLists.txt]: We should merge this first into `feature/ufs_dom` and then update the ufs-bundle PR.

File: CMakeLists.txt [hunk 5]
```diff
@@ -157,15 +161,15 @@ if(UFS_APP MATCHES "^(S2S)$")
...
+  ecbuild_bundle( PROJECT soca     GIT "https://github.com/jcsda-internal/soca.git"     BRANCH feature/ufs_clem )  # updated from develop on apr 18
```
Comment (climbfuji) [inline CMakeLists.txt]: same here

File: CMakeLists.txt [hunk 6]
```diff
@@ -157,15 +161,15 @@ if(UFS_APP MATCHES "^(S2S)$")
...
   add_dependencies(soca ufs-weather-model)
 elseif(UFS_APP MATCHES "^(NG-GODAS)$")
-  ecbuild_bundle( PROJECT soca     GIT "https://github.com/jcsda-internal/soca.git" BRANCH feature/ufs_dom )  # updated from develop on jan28
+  ecbuild_bundle( PROJECT soca     GIT "https://github.com/jcsda-internal/soca.git" BRANCH feature/ufs_clem )  # updated from develop on apr 18
```
Comment (climbfuji) [inline CMakeLists.txt]: and here

File: CMakeLists.txt [hunk 7]
```diff
@@ -157,15 +161,15 @@ if(UFS_APP MATCHES "^(S2S)$")
...
   add_dependencies(soca ufs-weather-model)
 elseif(UFS_APP MATCHES "^(ATM)$" OR UFS_APP MATCHES "^(ATMAERO)$")
   ecbuild_bundle( PROJECT femps    GIT "https://github.com/jcsda-internal/femps.git"    TAG 1.2.0 )
-  ecbuild_bundle( PROJECT fv3-jedi GIT "https://github.com/jcsda-internal/fv3-jedi.git" BRANCH feature/ufs_dom )  # updated from develop on jan28
+  ecbuild_bundle( PROJECT fv3-jedi GIT "https://github.com/jcsda-internal/fv3-jedi.git" BRANCH feature/ufs_clem_update )  # updated from develop on april13
```
Comment (climbfuji) [inline CMakeLists.txt]: and here

File: CMakeLists.txt [hunk 8]
```diff
@@ -204,7 +208,7 @@ branch_checkout (REPO_DIR_NAME ufo-data
 # same procedure for fv3-jedi-data
 find_branch_name(REPO_DIR_NAME fv3-jedi)
 if( NOT DEFINED ENV{LOCAL_PATH_JEDI_TESTFILES} AND NOT DEFINED ${GIT_TAG_FUNC} )
-  ecbuild_bundle( PROJECT fv3-jedi-data GIT "https://github.com/JCSDA-internal/fv3-jedi-data.git" BRANCH feature/ufs_dom ) # updated from develop on jan 28
+  ecbuild_bundle( PROJECT fv3-jedi-data GIT "https://github.com/JCSDA-internal/fv3-jedi-data.git" BRANCH feature/ufs_clem ) # updated from develop on apr 18
```
Comment (climbfuji) [inline CMakeLists.txt]: and lastly, here

**Review by climbfuji — CHANGES_REQUESTED** [review]

**Comment by cmgas** [comment]
@climbfuji When we merge the soca, fv3-jedi and fv3-jedi-data I will update the ufs-bundle PR

**Comment by climbfuji** [comment]
Would be nice to have a list of all PRs that this PR is waiting on in the description ...

**Comment by climbfuji** [comment]
Hmm, bad news at last. S2S compile failed on my macOS with:
```
[ 90%] Built target fv3jedi_addincrement.x
/Applications/Xcode.app/Contents/Developer/usr/bin/make  -f fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/build.make fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/depend
cd /Users/heinzell/scratch/ufs-bundle/build-debug-s2s && /usr/local/Cellar/cmake/3.23.1/bin/cmake -E cmake_depends "Unix Makefiles" /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13 /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13/fv3-jedi/src/mains /Users/heinzell/scratch/ufs-bundle/build-debug-s2s /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/fv3-jedi/src/mains /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make  -f fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/build.make fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/build
cd /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/etc && /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/bin/fv3jedi_addincrement.x --output-json-schema=fv3jedi_addincrement.x.schema.json
dyld[72844]: Library not loaded: libpiof.dylib
  Referenced from: /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/bin/fv3jedi_addincrement.x
  Reason: tried: 'libpiof.dylib' (no such file), '/usr/local/lib/libpiof.dylib' (no such file), '/usr/lib/libpiof.dylib' (no such file), '/Users/heinzell/scratch/ufs-bundle/build-debug-s2s/etc/libpiof.dylib' (no such file), '/usr/local/lib/libpiof.dylib' (no such file), '/usr/lib/libpiof.dylib' (no such file)
/bin/sh: line 1: 72844 Abort trap: 6           /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/bin/fv3jedi_addincrement.x --output-json-schema=fv3jedi_addincrement.x.schema.json
make[2]: *** [fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json] Error 134
make[1]: *** [fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/all] Error 2
make: *** [all] Error 2
```
This works with the current ufs-bundle develop.

**Comment by climbfuji** [comment]
> Hmm, bad news at last. S2S compile failed on my macOS with:
> 
> ```
> [ 90%] Built target fv3jedi_addincrement.x
> /Applications/Xcode.app/Contents/Developer/usr/bin/make  -f fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/build.make fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/depend
> cd /Users/heinzell/scratch/ufs-bundle/build-debug-s2s && /usr/local/Cellar/cmake/3.23.1/bin/cmake -E cmake_depends "Unix Makefiles" /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13 /Users/heinzell/scratch/ufs-bundle/ufs-bundle-clem-apr13/fv3-jedi/src/mains /Users/heinzell/scratch/ufs-bundle/build-debug-s2s /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/fv3-jedi/src/mains /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/DependInfo.cmake --color=
> /Applications/Xcode.app/Contents/Developer/usr/bin/make  -f fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/build.make fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/build
> cd /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/etc && /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/bin/fv3jedi_addincrement.x --output-json-schema=fv3jedi_addincrement.x.schema.json
> dyld[72844]: Library not loaded: libpiof.dylib
>   Referenced from: /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/bin/fv3jedi_addincrement.x
>   Reason: tried: 'libpiof.dylib' (no such file), '/usr/local/lib/libpiof.dylib' (no such file), '/usr/lib/libpiof.dylib' (no such file), '/Users/heinzell/scratch/ufs-bundle/build-debug-s2s/etc/libpiof.dylib' (no such file), '/usr/local/lib/libpiof.dylib' (no such file), '/usr/lib/libpiof.dylib' (no such file)
> /bin/sh: line 1: 72844 Abort trap: 6           /Users/heinzell/scratch/ufs-bundle/build-debug-s2s/bin/fv3jedi_addincrement.x --output-json-schema=fv3jedi_addincrement.x.schema.json
> make[2]: *** [fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json] Error 134
> make[1]: *** [fv3-jedi/src/mains/CMakeFiles/fv3jedi_addincrement.x.schema.json.dir/all] Error 2
> make: *** [all] Error 2
> ```
> 
> This works with the current ufs-bundle develop.

I believe that this is an error in the parallelio package itself. Not sure why it didn't show up until now. I'll do a bit more digging if we can work around it, but in the meanwhile we should start merging the upstream dependencies of this PR.

**Comment by climbfuji** [comment]
The three upstream dependencies have been merged.

**Review by shlyaeva — DISMISSED** [review]

**Comment by cmgas** [comment]
@climbfuji I updated all the branches to be `feature/ufs_dom`

**Review by climbfuji — APPROVED** [review]

**Review by shlyaeva — APPROVED** [review]
Did not retest since the only changes since the last test are in branch names, I expect this to work as it did before. 🎉

**Pull request merged**